### PR TITLE
Tremolo Fixes for #18173

### PR DESCRIPTION
**Pull request by asattely**
Resolves: https://github.com/musescore/MuseScore/issues/18173

- Relayout issue: The stem direction was based on its tremolo (like chords in beams) but the tremolo hadn't necessarily been laid out yet, so we were getting incorrect direction on score load.
- Beam styles: I don't know where they went, but they're back now
- Tremolo beams forced horizontal: There was a bug in the combined tremolo/beam placement code that was treating very specific combinations of multi-note chords in two-chord tremolos as beams with more than two notes. Weird. That's been fixed also
- Flipping manually-adjusted trems: Also fixed, the cause of this was the relayout issue above. Note that this is an engraving fix only--the properties panel will be addressed in a different PR
- Inserting time inside two-note trems: Any tremolo that is split in this way is removed and the two notes are retained.

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/engraving/libmscore/tremolo.cpp [hunk 1]
```diff
@@ -132,14 +132,23 @@ void Tremolo::createBeamSegments()
...
-    double offset = gapSp * spatium();
-    startAnchor.rx() += offset;
-    endAnchor.rx() -= offset;
-    startAnchor.ry() += offset * slope;
-    endAnchor.ry() -= offset * slope;
     BeamSegment* mainStroke = new BeamSegment(this);
+    double offset = gapSp * spatium();
+    if (_style == TremoloStyle::TRADITIONAL_ALTERNATE) {
+        mainStroke->line = LineF(startAnchor, endAnchor);
+        startAnchor.rx() += offset;
```
Comment (RomanPudashkin) [inline src/engraving/libmscore/tremolo.cpp:139]: Please don't use these rx() and ry(). Use setX and setY

File: src/engraving/layout/v0/beamtremololayout.cpp [hunk 2]
```diff
@@ -894,10 +894,12 @@ int BeamTremoloLayout::isSlopeConstrained(int startNote, int endNote) const
     // 0 to constrain to flat, 1 to constrain to 0.25, <0 for no constraint
     if (startNote == endNote) {
         return 0;
+    } else if (m_beamType == BeamType::TREMOLO) {
+        return -1;
```
Comment (RomanPudashkin) [inline src/engraving/layout/v0/beamtremololayout.cpp:898]: isSlopeConstrained is not a good name for this type of method. It should either return bool or have another name

File: src/engraving/layout/v0/beamtremololayout.cpp [hunk 3]
```diff
@@ -894,10 +894,12 @@ int BeamTremoloLayout::isSlopeConstrained(int startNote, int endNote) const
...
     }
     // if a note is more extreme than the endpoints, slope is 0
     // p.s. _notes is a sorted vector
```
Comment (RomanPudashkin) [inline src/engraving/layout/v0/beamtremololayout.cpp:901]: Wrong comment

**Comment by asattely** [comment]
All @RomanPudashkin's suggestions implemented, all of the engraving-related points of the original issue (not the UI one for tremolo stem direction) are now addressed!

**Review by RomanPudashkin — APPROVED** [review]

**Comment by RomanPudashkin** [comment]
@asattely thanks! Looks much better!

**Pull request merged**